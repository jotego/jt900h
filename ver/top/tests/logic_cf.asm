    ; Arithmetic operations that read & write to memory
    maxmode on
    relaxed on
    org 0
    ld a,0xbf    ; common header

    jp c,bad_end
    ld bc,0xaa55

    orcf 0,bc
    jp nc,bad_end
    or e,1

    ld a,1
    andcf a,bc
    jp c,bad_end
    or e,2

    lda xix,data
    xorcf 1,(xix)
    jp nc,bad_end
    or e,4

    ld a,0
    andcf a,(xix)
    jp c,bad_end

    include finish.inc
data:
    dw 0xcafe,0xbeef,0xffff,0xeeee,0xcccc
    end