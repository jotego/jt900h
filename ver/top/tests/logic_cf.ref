00000000 - 00000000 - 00000000 - 00000000
0000aa55 - 00000000 - 00000000 - 00000000
00000007 - 00000000 - 00000000 - 00000000
0000babe - 00000000 - 00000000 - 00000000
-- index registers --
00000051 - 00000000 - 00000000 - 00000100
SR = f800
