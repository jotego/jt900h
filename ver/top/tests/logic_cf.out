00000000 - 00000000 - 00000000 - 00000000
0000aa55 - 00000000 - 00000000 - 00000000
00000007 - 00000000 - 00000000 - 00000000
0000babe - 00000000 - 00000000 - 00000000
-- index registers --
0000004c - 00000000 - 00000000 - 00000000
SR = 00f8
